src/User.jsx

```diff
@@ -0,0 +1,322 @@
+    };
+  });
+
+  return {ref, isComponentVisible, setIsComponentVisible};
+}
+const User = ({address, setAddress, open, setOpen, toggleOpen}) => {
+
+  const [show, setShow] = useState(false);
+
+  const showModal = async e => {
+    e.preventDefault();
+    setShow(!show);
+  } 
+
+  const discordRef = useComponentVisible(false);
+  const metaMaskRef = useComponentVisible(false);
+  const emailRef = useComponentVisible(false);
+  const [loggingIn, setLoggingIn] = useState(false);
+  const [loginButtons, setLoginButtons] = useState(false);
+
+  const [walletData, setWalletData] = useState([]);
+  const [discordUrl, setDiscordUrl] = useState('');
+  const [loginError, setLoginError] = useState(null);
+
+  const userRef = useComponentVisible(false, ()=>{
+    setLoginError(false);
+    setDiscordUrl('');
+  });
+
+  useEffect(() => {
+    if(address && sessionStorage.getItem('temp_token')) {
+      fetchWalletData();
+    }
+  });
+
+  var popUp = null;
+  var iframe = null;
+  var walletMessenger = null;
+
+  const openPopup = async (token) => {
+    popUp = window.open(walletHost, '', "height=800,width=600");
+
+    window.addEventListener("message", (event) => {
+      if (event.origin !== walletHost)
+        return;
+    
+      if(event.data == 'received') {
+        sendMessage(token);
+      }
+    }, false);
+  }
+
+  const fetchWalletData = async () => {
+    iframe = window.open(walletHost, 'wallet')
+
+    window.addEventListener("message", (event) => {
+      if (event.origin !== walletHost)
+        return;
+
+      if(event.data == 'received') {
+        getKeys();
+      }
+    }, false);
+  }
+
+  // function for sending data to wallet
+  const sendDataToWallet = async (data) => {
+    walletMessenger = window.open(walletHost, 'walletMessenger')
+
+    window.addEventListener("message", (event) => {
+      if (event.origin !== walletHost)
+        return;
+    
+      if(event.data == 'received') {
+        sendData(data);
+      }
+    }, false);
+  }
+
+  const sendData = async (data) => {
+
+    // data format
+    // {'key': 'key-name', 'value': 'any-value'} 
+    var message = JSON.stringify({'temp_token': sessionStorage.getItem('temp_token'), 'token': address,
+    'data': data});
+    walletMessenger.postMessage(message, walletHost);
+
+    window.addEventListener("message", (event) => {
+      if (event.origin !== walletHost)
+        return;
+    
+      setWalletData(event.data.data);
+
+      window.removeEventListener('message', ()=>{});
+      walletMessenger.close();
+
+    }, false);
+  }
+
+  const getKeys = async () => {
+
+    iframe.postMessage(JSON.stringify({'temp_token': sessionStorage.getItem('temp_token'), 'token': address}), walletHost);
+
+    window.addEventListener("message", (event) => {
+      if (event.origin !== walletHost)
+        return;
+    
+      setWalletData(event.data.data);
+
+      window.removeEventListener('message', ()=>{});
```

Comment:
This isn't how you remove an events listener.

Reply:
> 

Fixed, updated removeEventListener code.